plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcClient.java

```diff
@@ -258,6 +260,15 @@ default Optional<TableScanRedirectApplicationResult> getTableScanRedirection(Con
 
     OptionalInt getMaxWriteParallelism(ConnectorSession session);
 
+    JdbcTableHandle updatedScanColumnsForMerge(ConnectorSession session, ConnectorTableHandle table, Optional<List<JdbcColumnHandle>> originalColumns, JdbcColumnHandle mergeRowIdColumnHandle);
```

Comment:
Thanks.
I moved it to the `JdbcPageSourceProvider` since I found we do not need it in the page sink. 
I updated the logic in `JdbcPageSourceProvider`, basicly we want to replace the merge id with its fields, then adapt to the upstream ` columns`, now it seems cleaner.